javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/ContextSpanProcessor.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.tooling;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.instrumentation.api.util.ContextSpanProcessorUtil;
+import io.opentelemetry.sdk.trace.ReadWriteSpan;
+import io.opentelemetry.sdk.trace.ReadableSpan;
+import io.opentelemetry.sdk.trace.SpanProcessor;
+import java.util.function.BiConsumer;
+
+/** A span processor that retrieves the actual processing function from context. */
+final class ContextSpanProcessor implements SpanProcessor {
```

Comment:
Instead of a span processor this should probably just be called from the instrumentation API so it can be used outside the javaagent. I think it could get rid of the weird bridging bytecode instrumentation too?

Reply:
Alternatively we would put this all in the SDK and call it directly there

Reply:
Calling it from instrumentation api is definitely an option. The downside would be that it wouldn't work for spans not created by instrumentation api, but that doesn't seem like a big drawback. The benefits of hooking into instrumentation api would be that we could also use an on end callback and also not needing to do the bridging.

Reply:
Cool, I agree the downside doesn't seem like too big of a deal given that the instrumentation API itself (or more precisely an incubator for now) is providing the functionality.

We'd likely want to put it in the SDK if we think it is appropriate to apply to all spans.

Reply:
Maybe we could put it into the `instrumentation-api-semconv` module? It kind of serves the incubator role.

Reply:
Changed to hook into instrumentation api instead of using a `SpanProcessor`. Using `instrumentation-api-semconv` isn't straightforward as `instrumentation-api-semconv` depends on `instrumentation-api` not the other way around.